runtime/commonMain/src/kotlinx/serialization/internal/SerialClassDescImpl.kt

```diff
@@ -31,6 +31,7 @@ public open class SerialClassDescImpl(
     // this array is only used when serializer is written by hand
     private var descriptors: MutableList<SerialDescriptor>? = null
     private var flags = BooleanArray(elementsCount)
+    internal val namesSet: Set<String> by lazy { names.toHashSet() }
```

Comment:
I start having a feeling that we'll need something like `SerialLocal` at some point :)

Reply:
We could leverage `DescriptorSchemaCache` from https://github.com/Kotlin/kotlinx.serialization/pull/597

Reply:
I'd rather avoid it for such a small task